### ISSUE TITLE
Compilation errors for weechat 1.0.1

### ISSUE THREAD

**Issue by stagnation** [khyperia/weechat-discord#31]
It could be that I have an old weechat version, I am running debian oldstable.
But here's my compilation errors. In the meantime I will look into updating my system

``` 
$ RUST_BACKTRACE=1 cargo build
   Compiling libc v0.2.28
   Compiling version_check v0.1.3
   Compiling weecord v0.1.0 (file:///home/spill/dotfiles/gits/weechat-discord)
   Compiling bitflags v0.7.0
   Compiling syn v0.11.11
error: failed to run custom build command for `weecord v0.1.0 (file:///home/spill/dotfiles/gits/weechat-discord)`
process didn't exit successfully: `/home/spill/dotfiles/gits/weechat-discord/target/debug/build/weecord-aa7e2c2fe3f6b7a6/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-unknown-linux-gnu = None
CC_x86_64_unknown_linux_gnu = None
HOST_CC = None
CC = None
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CFLAGS_x86_64-unknown-linux-gnu = None
CFLAGS_x86_64_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
PROFILE = Some("debug")
running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-m64" "-I" "/usr/include/weechat" "-Wall" "-Wextra" "-Werror" "-o" "/home/spill/dotfiles/gits/weechat-discord/target/debug/build/weecord-e9ca8906f351a336/out/src/weecord.o" "-c" "src/weecord.c"
cargo:warning=src/weecord.c: In function ‘wdc_hook_command’:
cargo:warning=src/weecord.c:66:35: error: macro "weechat_hook_command" passed 8 arguments, but takes just 7
cargo:warning=                               NULL);
cargo:warning=                                   ^
cargo:warning=src/weecord.c:59:10: error: ‘weechat_hook_command’ undeclared (first use in this function)
cargo:warning=   return weechat_hook_command(command,
cargo:warning=          ^
cargo:warning=src/weecord.c:59:10: note: each undeclared identifier is reported only once for each function it appears in
cargo:warning=src/weecord.c:46:30: error: unused parameter ‘command’ [-Werror=unused-parameter]
cargo:warning= wdc_hook_command(const char* command,
cargo:warning=                              ^
cargo:warning=src/weecord.c:47:30: error: unused parameter ‘description’ [-Werror=unused-parameter]
cargo:warning=                  const char* description,
cargo:warning=                              ^
cargo:warning=src/weecord.c:48:30: error: unused parameter ‘args’ [-Werror=unused-parameter]
cargo:warning=                  const char* args,
cargo:warning=                              ^
cargo:warning=src/weecord.c:49:30: error: unused parameter ‘args_description’ [-Werror=unused-parameter]
cargo:warning=                  const char* args_description,
cargo:warning=                              ^
cargo:warning=src/weecord.c:50:30: error: unused parameter ‘completion’ [-Werror=unused-parameter]
cargo:warning=                  const char* completion,
cargo:warning=                              ^
cargo:warning=src/weecord.c:51:30: error: unused parameter ‘pointer’ [-Werror=unused-parameter]
cargo:warning=                  const void* pointer,
cargo:warning=                              ^
cargo:warning=src/weecord.c:52:24: error: unused parameter ‘callback’ [-Werror=unused-parameter]
cargo:warning=                  int (*callback)(const void* pointer,
cargo:warning=                        ^
cargo:warning=src/weecord.c: In function ‘wdc_buffer_new’:
cargo:warning=src/weecord.c:123:71: error: macro "weechat_buffer_new" passed 7 arguments, but takes just 5
cargo:warning=     name, input_callback, pointer, NULL, close_callback, pointer, NULL);
cargo:warning=                                                                       ^
cargo:warning=src/weecord.c:122:10: error: ‘weechat_buffer_new’ undeclared (first use in this function)
cargo:warning=   return weechat_buffer_new(
cargo:warning=          ^
cargo:warning=src/weecord.c:112:28: error: unused parameter ‘name’ [-Werror=unused-parameter]
cargo:warning= wdc_buffer_new(const char* name,
cargo:warning=                            ^
cargo:warning=src/weecord.c:113:28: error: unused parameter ‘pointer’ [-Werror=unused-parameter]
cargo:warning=                const void* pointer,
cargo:warning=                            ^
cargo:warning=src/weecord.c:114:22: error: unused parameter ‘input_callback’ [-Werror=unused-parameter]
cargo:warning=                int (*input_callback)(const void* pointer,
cargo:warning=                      ^
cargo:warning=src/weecord.c:118:22: error: unused parameter ‘close_callback’ [-Werror=unused-parameter]
cargo:warning=                int (*close_callback)(const void* pointer,
cargo:warning=                      ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_fd’:
cargo:warning=src/weecord.c:159:62: error: macro "weechat_hook_fd" passed 7 arguments, but takes just 6
cargo:warning=   return weechat_hook_fd(fd, 1, 0, 0, callback, pointer, NULL);
cargo:warning=                                                              ^
cargo:warning=src/weecord.c:159:10: error: ‘weechat_hook_fd’ undeclared (first use in this function)
cargo:warning=   return weechat_hook_fd(fd, 1, 0, 0, callback, pointer, NULL);
cargo:warning=          ^
cargo:warning=src/weecord.c:155:17: error: unused parameter ‘fd’ [-Werror=unused-parameter]
cargo:warning= wdc_hook_fd(int fd,
cargo:warning=                 ^
cargo:warning=src/weecord.c:156:25: error: unused parameter ‘pointer’ [-Werror=unused-parameter]
cargo:warning=             const void* pointer,
cargo:warning=                         ^
cargo:warning=src/weecord.c:157:19: error: unused parameter ‘callback’ [-Werror=unused-parameter]
cargo:warning=             int (*callback)(const void* pointer, void* data, int fd))
cargo:warning=                   ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_completion’:
cargo:warning=src/weecord.c:243:67: error: macro "weechat_hook_completion" passed 5 arguments, but takes just 4
cargo:warning=     completion_item, description, callback, callback_pointer, NULL);
cargo:warning=                                                                   ^
cargo:warning=src/weecord.c:242:10: error: ‘weechat_hook_completion’ undeclared (first use in this function)
cargo:warning=   return weechat_hook_completion(
cargo:warning=          ^
cargo:warning=src/weecord.c:233:33: error: unused parameter ‘completion_item’ [-Werror=unused-parameter]
cargo:warning= wdc_hook_completion(const char* completion_item,
cargo:warning=                                 ^
cargo:warning=src/weecord.c:234:33: error: unused parameter ‘description’ [-Werror=unused-parameter]
cargo:warning=                     const char* description,
cargo:warning=                                 ^
cargo:warning=src/weecord.c:235:33: error: unused parameter ‘callback_pointer’ [-Werror=unused-parameter]
cargo:warning=                     const void* callback_pointer,
cargo:warning=                                 ^
cargo:warning=src/weecord.c:236:27: error: unused parameter ‘callback’ [-Werror=unused-parameter]
cargo:warning=                     int (*callback)(const void*,
cargo:warning=                           ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_command’:
cargo:warning=src/weecord.c:67:1: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning= }
cargo:warning= ^
cargo:warning=src/weecord.c: In function ‘wdc_buffer_new’:
cargo:warning=src/weecord.c:124:1: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning= }
cargo:warning= ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_fd’:
cargo:warning=src/weecord.c:160:1: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning= }
cargo:warning= ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_completion’:
cargo:warning=src/weecord.c:244:1: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning= }
cargo:warning= ^
cargo:warning=cc1: all warnings being treated as errors
exit code: 1


command did not execute successfully, got: exit code: 1



--- stderr
thread 'main' panicked at 'explicit panic', /home/spill/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.51/src/lib.rs:1239
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:355
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:371
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:549
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:511
   6: gcc::fail
             at /home/spill/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.51/src/lib.rs:1239
   7: gcc::run
             at /home/spill/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.51/src/lib.rs:1183
   8: gcc::Config::compile_object
             at /home/spill/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.51/src/lib.rs:525
   9: gcc::Config::compile_objects
             at /home/spill/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.51/src/lib.rs:491
  10: gcc::Config::compile
             at /home/spill/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.51/src/lib.rs:440
  11: build_script_build::main
             at ./build.rs:18
  12: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:98
  13: std::rt::lang_start
             at /checkout/src/libstd/panicking.rs:433
             at /checkout/src/libstd/panic.rs:361
             at /checkout/src/libstd/rt.rs:59
  14: main
  15: __libc_start_main
  16: <unknown>

Build failed, waiting for other jobs to finish...
error: build failed
```

```
spill@spilldesk:~/dotfiles/gits/weechat-discord$ git rev-parse HEAD
cf9f593648179976696dffe6e95536b7274a220b

spill@spilldesk:~/dotfiles/gits/weechat-discord$ gcc --version
gcc (Debian 4.9.2-10) 4.9.2
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

spill@spilldesk:~/dotfiles/gits/weechat-discord$ rustc --version
rustc 1.19.0-nightly (28fd1e519 2017-05-27)

spill@spilldesk:~/dotfiles/gits/weechat-discord$ cargo --version
cargo 0.20.0-nightly (38ca9b702 2017-05-14)

spill@spilldesk:~/dotfiles/gits/weechat-discord$ weechat --version
1.0.1
```

**Comment by stagnation** [comment]
Closing, this seems to be solved when updating to weechat 1.6.1.

**Comment by khyperia** [comment]
Correct in that your weechat is too old - this is a duplicate of #23, #24, and a comment of #15
